pkg/controller/argocd/keycloak.go

```diff
@@ -388,6 +452,181 @@ func newKeycloakTemplate(cr *argoprojv1a1.ArgoCD) (template.Template, error) {
 	return tmpl, err
 }
 
+func newKeycloakIngress(cr *argoprojv1a1.ArgoCD) *extv1beta1.Ingress {
+
+	return &extv1beta1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      defaultKeycloakIdentifier,
+			Namespace: cr.Namespace,
+		},
+		Spec: extv1beta1.IngressSpec{
+			TLS: []extv1beta1.IngressTLS{
+				{
+					Hosts: []string{keycloakIngressHost},
+				},
+			},
+			Rules: []extv1beta1.IngressRule{
+				{
+					Host: keycloakIngressHost,
+					IngressRuleValue: extv1beta1.IngressRuleValue{
+						HTTP: &extv1beta1.HTTPIngressRuleValue{
+							Paths: []extv1beta1.HTTPIngressPath{
+								{
+									Backend: extv1beta1.IngressBackend{
+										ServiceName: defaultKeycloakIdentifier,
+										ServicePort: intstr.FromInt(int(httpPort)),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func newKeycloakService(cr *argoprojv1a1.ArgoCD) *corev1.Service {
+
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      defaultKeycloakIdentifier,
+			Namespace: cr.Namespace,
+			Labels: map[string]string{
+				"app": defaultKeycloakIdentifier,
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{Name: "http", Port: httpPort, TargetPort: intstr.FromInt(int(httpPort))},
+			},
+			Selector: map[string]string{
+				"app": defaultKeycloakIdentifier,
+			},
+			Type: "LoadBalancer",
+		},
+	}
+}
+
+func newKeycloakDeployment(cr *argoprojv1a1.ArgoCD) *k8sappsv1.Deployment {
+
+	var replicas int32 = 1
+	return &k8sappsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      defaultKeycloakIdentifier,
+			Namespace: cr.Namespace,
+			Annotations: map[string]string{
+				"argocd.argoproj.io/realm-created": "false",
+			},
+			Labels: map[string]string{
+				"app": defaultKeycloakIdentifier,
+			},
+		},
+		Spec: k8sappsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": defaultKeycloakIdentifier,
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": defaultKeycloakIdentifier,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  defaultKeycloakIdentifier,
+							Image: defaultKeycloakImage,
```

Comment:
We should get the keycloak image in consistent way.  (i.e. in this order of precedence: get it from CR, Env var, and default).   We should also inherit proxy variables from operator.

Reply:
ACK. Made required changes.